TempDirectory.Test/TempDirectoryBuilderTest.cs

```diff
@@ -0,0 +1,115 @@
+using System.Collections.Generic;
+using System.IO;
+using Messerli.TempDirectory;
+using Xunit;
+using JetBrains.Annotations;
+
+namespace TempDirectory.Test
+{
+    public class TempDirectoryBuilderTest
+    {
+        [Theory]
+        [MemberData(nameof(GetTempDirectoryBuilderConfigurations))]
+        public void CreatesTempDirectory(TempDirectoryBuilder tempDirectoryBuilder)
+        {
+            using (var tempDirectory = tempDirectoryBuilder.Create())
+            {
+                Assert.True(Directory.Exists(tempDirectory.FullName));
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(GetTempDirectoryBuilderConfigurations))]
+        public void DeletesTempDirectory(TempDirectoryBuilder tempDirectoryBuilder)
+        {
+            var tempDirectory = tempDirectoryBuilder.Create();
+            Assert.True(Directory.Exists(tempDirectory.FullName));
+            tempDirectory.Dispose();
+
+            Assert.False(Directory.Exists(tempDirectory.FullName));
+        }
+
+        [Theory]
+        [MemberData(nameof(GetTempDirectoryBuilderConfigurations))]
+        public void FullNameContainsName(TempDirectoryBuilder tempDirectoryBuilder)
+        {
+            using (var tempDirectory = tempDirectoryBuilder.Create())
+            {
+                Assert.Contains(tempDirectory.Name, tempDirectory.FullName);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(GetTempDirectoryBuilderConfigurations))]
+        public void FullNameIsNotName(TempDirectoryBuilder tempDirectoryBuilder)
+        {
+            using (var tempDirectory = tempDirectoryBuilder.Create())
+            {
+                Assert.NotEqual(tempDirectory.FullName, tempDirectory.Name);
+            }
+        }
+
+        [Fact]
+        public void PrefixIsInName()
+        {
+            using (var tempDirectory = new TempDirectoryBuilder().Prefix(Prefix).Create())
+            {
+                Assert.Contains(Prefix, tempDirectory.Name);
+            }
+        }
+
+        [Fact]
+        public void SuffixIsInName()
+        {
+            using (var tempDirectory = new TempDirectoryBuilder().Suffix(Suffix).Create())
+            {
+                Assert.Contains(Suffix, tempDirectory.Name);
+            }
+        }
+
+        [Fact]
+        public void SeparatorsAreInName()
+        {
+            using (var tempDirectory = new TempDirectoryBuilder()
+                .Prefix(Prefix)
+                .PrefixSeparator(PrefixSeparator)
+                .Suffix(Prefix)
+                .SuffixSeparator(SuffixSeparator)
+                .Create())
+            {
+                Assert.Contains(PrefixSeparator, tempDirectory.Name);
```

Comment:
Same as the two comments above.

Reply:
Are you sure? I don't think the separators should be at the start or end of the string.